### PR TITLE
Default to categorical columns and KNN trainers 

### PR DESCRIPTION
**Pull request by Erin007**
- default to categorical data type if not already set 
- move column data type confirmation to TrainingSettings component, along with extracted slider to reserve data for testing
- skip specify columns panel at the beginning of the flow
-  include column type dropdown in single column data visualization panel 
- put all the "advanced controls" in a TrainingSettingsAdvanced component that's current not used  
- default to KNN and use data type of label column to set classification or regression if selectedTrainer isn't explicitly set 

https://user-images.githubusercontent.com/12300669/109667958-ad437480-7b3e-11eb-9e46-e3ffe78f9b25.mov

### CODE REVIEW

File: src/UIComponents/TrainingSettings.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  render() {
+    return (
+      <div id="select-trainer" style={styles.panel}>
+        <div>
+          <div style={styles.largeText}>Are you ready to train A.I. Bot?</div>
+          <div style={styles.largeText}>
+            A.I. Bot will predict {this.props.labelColumn} based on{" "}
+            {this.props.selectedFeatures.join(", ")}.
+          </div>
```
Comment (breville) [inline src/UIComponents/TrainingSettings.jsx:23]: We should just have one `largeText` and then smaller text for the second sentence.

File: src/UIComponents/TrainingSettings.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          <PercentReserveSlider />
+          <ColumnTypeSelector />
```
Comment (breville) [inline src/UIComponents/TrainingSettings.jsx:25]: If we have loaded metadata, I don't think we should show this here.

File: src/UIComponents/ColumnInspector.jsx [hunk 3]
```diff
@@ -77,9 +77,21 @@ class ColumnInspector extends Component {
...
+                  value={currentColumnData.dataType}
+                >
+                  {Object.values(ColumnTypes).map((option, index) => {
+                    return (
+                      <option key={index} value={option}>
+                        {option}
+                      </option>
+                    );
+                  })}
+                </select>
```
Comment (breville) [inline src/UIComponents/ColumnInspector.jsx:94]: If we have preloaded metadata, I think we should just show the data type here without allowing the user to change it.

File: src/App.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 import SelectDataset from "./UIComponents/SelectDataset";
-import SpecifyColumns from "./UIComponents/SpecifyColumns";
-import SpecifyColumnsInfo from "./UIComponents/SpecifyColumnsInfo";
```
Comment (breville) [inline src/App.js:5]: Should we delete these files too, then?

File: src/UIComponents/ColumnInspector.jsx [hunk 5]
```diff
@@ -136,7 +153,8 @@ class ColumnInspector extends Component {
 export default connect(
   state => ({
     currentColumnData: getCurrentColumnData(state),
-    rangesByColumn: getRangesByColumn(state)
+    rangesByColumn: getRangesByColumn(state),
+    hideSpecifyColumns: state.mode.hideSpecifyColumns
```
Comment (breville) [inline src/UIComponents/ColumnInspector.jsx:157]: This hits an exception when `mode` is `null` which happens in our default standalone invocation.

Comment (breville) [reply src/UIComponents/ColumnInspector.jsx:157]: (We guard against it [here](https://github.com/code-dot-org/ml-playground/pull/83/files#diff-6e02291d2b49a002990985f9cb58187da3ad794009887859ccce802714a852ddL32), for example.)

File: src/App.js (hunk 4 above)
Comment (Erin007) [reply src/App.js:5]: done!

File: src/UIComponents/ColumnInspector.jsx (hunk 5 above)
Comment (Erin007) [reply src/UIComponents/ColumnInspector.jsx]: done! Thanks for the tip.

**Review by breville — APPROVED** [review]

**Pull request merged**